src/sidebar/services/annotations.js

```diff
@@ -241,11 +244,14 @@ export class AnnotationsService {
       this._store.annotationSaveFinished(annotation);
     }
 
-    Object.keys(annotation).forEach(key => {
-      if (key[0] === '$') {
-        savedAnnotation[key] = annotation[key];
+    // Copy local/internal fields from the original annotation to the saved
+    // version.
+    for (let [key, value] of Object.entries(annotation)) {
+      if (key.startsWith('$')) {
+        const fields = /** @type {Record<string, any>} */ (savedAnnotation);
```

Comment:
Yes. If you try to do `savedAnnotation[key]` directly, TS complains that `Annotation` can't be indexed with an arbitrary string key. If you specifically cast `key` to a `keyof Annotation`, TS still complains because it can't be sure that the value is appropriate for a particular key. So here we're just casting to a general `Record` as an escape-hatch.